### PR TITLE
Update deps and modify code for dart-sass etc

### PR DESCRIPTION
**Pull request by markstuart**
Updated to latest deps for build pipeline. This removes the reliance on node-sass and moves to Dart sass instead.

The `npm run build` command works, and I've gone through and fixed a bunch of deprecations/errors.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ const autoprefixer = require('gulp-autoprefixer')
 const rename = require('gulp-rename')
 const copy = require('gulp-copy')
 const clean = require('gulp-clean')
-const sass = require('gulp-sass')
+const sass = require('gulp-sass')(require('sass'))
```
Comment (markstuart) [inline gulpfile.js:9]: This is the pattern to tell 'gulp-sass' to use the 'sass' dep instead of 'node-sass'.

File: package.json [hunk 2]
```diff
@@ -25,31 +25,35 @@
...
   },
   "browserslist": [
     "defaults",
     ">0.1%"
-  ]
+  ],
+  "overrides": {
+    "svg2ttf": "6.0.3",
+    "webpack": "^5.0"
+  }
```
Comment (markstuart) [inline package.json:58]: I bumped everything to the latest available versions and worked through `npm audit` fixes as much as I could. These overrides force modules to use these versions if they have them as a sub-dependency, similar to yarn's resolutions.

File: src/1-helpers/_breakpoints.scss [hunk 3]
```diff
@@ -10,7 +10,7 @@ $breakpoints: (
 /// The largest named breakpoint in which to include print as a media type
 $print-breakpoint: lg !default;
 $zero-breakpoint: default !default;
-$breakpoints-keys: map-to-list($breakpoints, 'keys');
+$breakpoints-keys: map-keys($breakpoints);
```
Comment (markstuart) [inline src/1-helpers/_breakpoints.scss:13]: map-to-list no longer exists

File: src/1-helpers/_value.scss [hunk 4]
```diff
@@ -57,7 +59,7 @@
   // If the base font size is a %, then multiply it by 16px
   // This is because 100% font size = 16px in most all browsers
   @if unit($base) == '%' {
-    $base: ($base / 100%) * 16px;
+    $base: calc($base / 100%) * 16px;
```
Comment (markstuart) [inline src/1-helpers/_value.scss:62]: Lots of deprecations about division needing to happen inside a `calc` function

File: src/5-navigation/menu/_index.scss [hunk 5]
```diff
@@ -128,14 +128,16 @@ $menu-item-active-class: 'current' !default;
...
-        $label: '2-column'
-      ) {
-        @include menu-direction(horizontal);
-        & > li {
-          width: 50%;
+      @if $breakpoint != 'default' {
+        @include selector(
+          $prefix: '&',
+          $breakpoint: $breakpoint,
+          $label: '2-column'
```
Comment (markstuart) [inline src/5-navigation/menu/_index.scss:135]: This crazy sass function generates prefixed classes for each defined breakpoint, like 'md-2-column', 'lg-2-column', but it was also creating one as just '2-column'... which is an invalid class name in css, can't start with a number.

File: package.json (hunk 2 above)
Comment (markstuart) [reply package.json:60]: Since the above comment, I've moved the repo to yarn... the resolutions and auditing just makes more sense.

**Comment by niall-mediasuite** [comment]
Sorry I didn't get around to this yesterday. Nice, looks like some good progress on updating some of the older parts. I've had a skim but its a lot of stuff I don't really have any info on. Let me know if you need something more thorough, preferably ASAP so I can fit it in today.

**Pull request merged**